From 36e64abed2209dcd3ec8b33cf9f1e605b05c593e Mon Sep 17 00:00:00 2001
From: "Wu, Qiming" <qi-ming.wu@intel.com>
Date: Mon, 23 Jul 2018 05:47:02 +0200
Subject: [PATCH] Merge pull request #88 in SW_UGW/linux from
 DRVLIB_SW-849-dp-enable-config_ltq_datapath_aca_csum_workaround to master

* commit '6353a226d52da8088186fd5639593d38e1593c58':
  Jira DRVLIB_SW-849: dp fix compilation after enable CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND by default
---
 drivers/net/ethernet/lantiq/cqm/grx500/cbm.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
index a69d84725f12..5b6495331842 100644
--- a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
+++ b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
@@ -330,10 +330,8 @@ s32 pmac, u32 flags)
 			if ((ptr->pmac == pmac) &&
 			    (!(ptr->egp_type & DP_F_MPE_ACCEL)) &&
 			    (!(ptr->egp_type & DP_F_DIRECTPATH_RX)) &&
-			#ifndef CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND
-				(!(ptr->egp_type & DP_F_CHECKSUM))
-			#endif
-			) {
+			    (IS_ENABLED(CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND)
+			     || (!(ptr->egp_type & DP_F_CHECKSUM)))) {
 				spin_unlock_irqrestore(&cbm_port_mapping,
 						       lock_flags);
 				return ptr;
@@ -412,7 +410,7 @@ struct cbm_egp_map epg_lookup_table[] = {
 	{17,	6,			 DP_F_FAST_ETH_LAN},
 	{18,	CBM_PMAC_NOT_APPL, DP_F_FAST_DSL_DOWNSTREAM},
 	{19,	15,			 DP_F_FAST_ETH_WAN},
-	#ifdef CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND
+	#if IS_ENABLED(CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND)
 	{20,	CBM_PMAC_DYNAMIC, DP_F_CHECKSUM},
 	#else
 	{20,	CBM_PMAC_NOT_APPL, DP_F_CHECKSUM},
@@ -3144,7 +3142,7 @@ s32 direct_dp_enable(u32 port_id, u32 flags, u32 type)
 	return CBM_SUCCESS;
 }
 
-#ifdef CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND
+#if IS_ENABLED(CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND)
 s32 checksum_dp_enable(u32 port_id, u32 flags, u32 type)
 {
 	int j, queue_number;
@@ -3283,7 +3281,7 @@ dp_enable(
 		if ((local_entry->egp_type == DP_F_DIRECT) && (flags & CBM_PORT_F_DISABLE))
 			goto UNLOCK;
 	}
-	#ifdef CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND
+	#if IS_ENABLED(CONFIG_LTQ_DATAPATH_ACA_CSUM_WORKAROUND)
 	else if (local_entry && (local_entry->egp_type == DP_F_CHECKSUM))
 		checksum_dp_enable(port_id, flags, local_entry->egp_type);
 	#endif
